scipy/stats/kde.py

```diff
@@ -204,23 +219,45 @@ def evaluate(self, points):
                 raise ValueError(msg)
 
         result = zeros((m,), dtype=float)
+        result_l = zeros((m,), dtype=float)
+        result_u = zeros((m,), dtype=float)
 
         if m >= self.n:
             # there are more points than data, so loop over data
             for i in range(self.n):
-                diff = self.dataset[:, i, newaxis] - points
-                tdiff = dot(self.inv_cov, diff)
-                energy = sum(diff*tdiff,axis=0) / 2.0
-                result = result + exp(-energy)
+                diff_0 = self.dataset[:, i, newaxis] - points
+                tdiff_0 = dot(self.inv_cov, diff_0)
+                energy_0 = sum(diff_0 * tdiff_0, axis=0) / 2.0
+                result = result + exp(-energy_0)
+                if lb is not None:
+                    diff_l = self.dataset + points[:, i, newaxis] - 2*lb
+                    tdiff_l = dot(self.inv_cov, diff_l)
+                    energy_l = sum(diff_l * tdiff_l, axis=0) / 2.0
+                    result_l[i] = sum(exp(-energy_l))
+                if ub is not None:
+                    diff_u = self.dataset + points[:, i, newaxis] - 2*ub
+                    tdiff_u = dot(self.inv_cov, diff_u)
+                    energy_u = sum(diff_u * tdiff_u, axis=0) / 2.0
+                    result_u[i] = sum(exp(-energy_u))
         else:
             # loop over points
             for i in range(m):
-                diff = self.dataset - points[:, i, newaxis]
-                tdiff = dot(self.inv_cov, diff)
-                energy = sum(diff * tdiff, axis=0) / 2.0
-                result[i] = sum(exp(-energy), axis=0)
-
-        result = result / self._norm_factor
+                diff_0 = self.dataset - points[:, i, newaxis]
+                tdiff_0 = dot(self.inv_cov, diff_0)
+                energy_0 = sum(diff_0 * tdiff_0, axis=0) / 2.0
+                result[i] = sum(exp(-energy_0), axis=0)
+                if lb is not None:
+                    diff_l = self.dataset + points[:, i, newaxis] - 2*lb
+                    tdiff_l = dot(self.inv_cov, diff_l)
+                    energy_l = sum(diff_l * tdiff_l, axis=0) / 2.0
+                    result_l[i] = sum(exp(-energy_l), axis=0)
```

Comment:
one possible change: this could be added inplace to `result[i]` instead of using additional temporary arrays, AFAICS.